components/Users/MealEntriesPerEmail.jsx

```diff
@@ -0,0 +1,46 @@
+// reusable component that calls /api/mealEntriesPerEmail api to get  meal entries per email
+
+import axios from "axios";
+import React, { useState } from "react";
+import useSWR, { mutate, useSWRConfig } from "swr";
+
+const fetcherForUserInfo = (url, userEmail) =>
+  axios
+    .get(url, { params: { userEmail: userEmail } })
+    .then((response) => response.data);
+
+export const MealEntriesPerEmail =  (props) => {
+    const { data:mealEntriesPerEmail, error,mutate } = useSWR( ["/api/mealEntriesPerEmail", props.sessionEmail], fetcherForUserInfo);
```

Comment:
Move `/api/mealEntriesPerEmail` url to a separate file (for centralized) like - `lib/urls/meal.js`